### PR TITLE
new data format implemented in event_reader

### PR DESCRIPTION
**Pull request by acolijn**
I have rewritten the event_reader.py class. It now reads the new data format. It can return an numpy array with all the event data.

### CODE REVIEW

File: optosim/simulation/event_reader.py [hunk 1]
```diff
@@ -71,172 +61,64 @@ def __init__(self, filenames):
...
-        None
-
-        Returns
-        -------
-        event_data : dict
-            The event data
-
-        A.P. Colijn
-        """
+        print("huh config = ", self.config)
```
Comment (marjoleinvannuland) [inline optosim/simulation/event_reader.py]: What's this?

File: optosim/simulation/event_reader.py [hunk 2]
```diff
@@ -71,172 +61,64 @@ def __init__(self, filenames):
...
-        Returns
-        -------
-        event_data : dict
-            The event data
-
-        A.P. Colijn
-        """
+        print("huh config = ", self.config)        
 
         if self.config['data_type_version'] == 1.0:
```
Comment (cfuselli) [inline optosim/simulation/event_reader.py]: This gives an error if the data_type_version is not in the config. As old data does not have it, I suggest updating with: 
`
        if self.config.get("data_type_version", 1.0) == 1.0:
`

File: optosim/simulation/event_reader.py [hunk 3]
```diff
@@ -250,9 +132,84 @@ def print_event(self, event):
 
         A.P. Colijn
         """
+        event = self.get_event(n)
         print(event)
 
-    def show_event(self, event):
+    # def show_event(self, n):
```
Comment (marjoleinvannuland) [inline optosim/simulation/event_reader.py]: Is this used or can it be deleted?

**Review by cfuselli — CHANGES_REQUESTED** [review]
See requested changes

File: optosim/simulation/event_reader.py [hunk 4]
```diff
@@ -71,172 +61,64 @@ def __init__(self, filenames):
...
-                event_dataset = self.file["events"][self.event_names[self.event_index]]
-                event_data = {key: event_dataset[key][()] for key in event_dataset.keys()}
-                return event_data
+            # throw an error that this version is obsolete
+            raise ValueError("Data type version 1.0 is obsolete. Please use version 2.0")
+
+        # Load data from all files
+        self.data_dict = self._load_data(filenames)
+        self.number_of_files = len(filenames)
+        self.num_events = len(next(iter(self.data_dict.values())))  # assuming all datasets have the same length
```
Comment (marjoleinvannuland) [inline optosim/simulation/event_reader.py:76]: Is there a better way to do this, without assuming all datasets are the same length?

**Review by cfuselli — APPROVED** [review]
ok better. good for me!

**Pull request merged**